erpnext/hr/doctype/leave_policy_assignment/leave_policy_assignment.js

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2020, Frappe Technologies Pvt. Ltd. and contributors
+// For license information, please see license.txt
+
+frappe.ui.form.on('Leave Policy Assignment', {
+	onload: function(frm){
+		frm.ignore_doctypes_on_cancel_all = ["Leave Ledger Entry"];
+	},
+
+	refresh: function(frm) {
+		if(frm.doc.docstatus === 1 && frm.doc.already_allocated === 0){
+			frm.add_custom_button(__("Grant Leave"), function() {
+
+				frappe.call({
+					doc: frm.doc,
+					method: "grant_leave_alloc_for_employee",
+					callback: function(r) {
+						let leave_allocations = r.message;
+
+						let msg = __("Leaves has been granted successfully");
+						msg += "<br><table class='table table-bordered'>";
+						msg += "<tr><th>"+__('Leave Type')+"</th><th>"+__("Leave Allocation")+"</th><th>"+__("Leaves Granted")+"</th><tr>";
+						for (let key in leave_allocations){
+
+							msg += "<tr><th>"+key+"</th><td>"+leave_allocations[key]["name"]+"</td><td>"+leave_allocations[key]["leaves"]+"</td></tr>";
+						}
+						msg += "</table>";
```

Comment:
Move to a new function `get_success_message`

Reply:
done